connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -166,11 +176,14 @@ protected void close() {
                 log.warn("Could not close producer", t);
             }
         }
-        try {
-            transformationChain.close();
-        } catch (Throwable t) {
-            log.warn("Could not close transformation chain", t);
+        if (admin != null) {
+            try {
+                admin.close(Duration.ofSeconds(30));
+            } catch (Throwable t) {
+                log.warn("Failed to close admin client on time", t);
+            }
```

Comment:
We don't do that with other clients or in `closeQuietly`
I'd say we don't need to because although not required by `AutoCloseable` the admin implementation of `close` seems idempotent.